flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaConnectorOptions.java

```diff
@@ -195,12 +196,6 @@
                                                     "custom class name (use custom FlinkKafkaPartitioner subclass)"))
                                     .build());
 
-    public static final ConfigOption<SinkSemantic> SINK_SEMANTIC =
```

Comment:
@twalthr I wonder if we can safely ignore it or what would be the preferred option here?

Reply:
Since the values remain the same we should at least add `sink.semantic` as a deprecated fallback key for now. The enum itself only became public during 1.14 anyway so I would expect that that isn't an issue.

Reply:
:+1: to fallback key `sink.semantic`. Are values compatible?